### PR TITLE
Fix RouteSelector check

### PR DESCRIPTION
**Pull request by sam-nguyen7**
- Fix RouteSelector check which causes non-default ingress to be deleted
- Add error check fro creation of non-default ingress

Addresses OHSS-62

### CODE REVIEW

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ parameters:
 metadata:
   name: selectorsyncset-template
 objects:
-- apiVersion: hive.openshift.io/v1alpha1
```
Comment (fahlmant) [inline hack/olm-registry/olm-artifacts-template.yaml:18]: What's the reason for this change?

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go [hunk 2]
```diff
@@ -273,7 +273,7 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 			log.Error(err, "cannot get infrastructure name")
 			return reconcile.Result{}, err
 		}
-		extNLBName := infrastructureName + "-test"
+		extNLBName := infrastructureName + "-ext"
```
Comment (fahlmant) [inline pkg/controller/publishingstrategy/publishingstrategy_controller.go:276]: Reason for change? Does this affect the logic changing below?

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go [hunk 3]
```diff
@@ -545,12 +559,14 @@ func isOnCluster(publishingStrategyIngress *cloudingressv1alpha1.ApplicationIngr
 		return false
 	}
 	if publishingStrategyIngress.RouteSelector.MatchLabels != nil {
```
Comment (fahlmant) [inline pkg/controller/publishingstrategy/publishingstrategy_controller.go:561]: Should this be an or instead of nested?

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 1 above)
Comment (cblecker) [reply hack/olm-registry/olm-artifacts-template.yaml:18]: We should be using hive/v1 everywhere by now

Comment (fahlmant) [reply hack/olm-registry/olm-artifacts-template.yaml:18]: Has the operator been tested with hive v1?

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go (hunk 2 above)
Comment (sam-nguyen7) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go:276]: this was found by Will Gordon. Very small change. Does not change logic below.

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 1 above)
Comment (sam-nguyen7) [reply hack/olm-registry/olm-artifacts-template.yaml:18]: https://github.com/openshift/cloud-ingress-operator/pull/48 this PR has been merged awhile back. It was just never generated. afaik it's OK be be rolled out

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go (hunk 2 above)
Comment (fahlmant) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go:276]: Should this be done in a seperate PR or commit then?

File: pkg/controller/publishingstrategy/publishingstrategy_controller.go (hunk 3 above)
Comment (lisa) [reply pkg/controller/publishingstrategy/publishingstrategy_controller.go:561]: I agree with @fahlmant on this one. Can this be re-worked to avoid the nested conditionals?

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 1 above)
Comment (fahlmant) [reply hack/olm-registry/olm-artifacts-template.yaml:18]: IMO should be in a different commit/PR since if you revert this PR this will be reverted as well. But I won't block PR on this

Comment (lisa) [reply hack/olm-registry/olm-artifacts-template.yaml:18]: I agree with Taylor here, I wouldn't 👎 but if we can easily split it out, we should.

**Comment by lisa** [comment]
@sam-nguyen7 Before we promote this let's talk. We might need to revert #63 if the behaviour can't be quickly understood.
